packages/twenty-front/src/modules/settings/data-model/fields/forms/relation/hooks/useRelationSettingsFormInitialValues.ts

```diff
@@ -27,17 +27,26 @@ export const useRelationSettingsFormInitialValues = ({
     [fieldMetadataItem, getRelationMetadata],
   ) ?? {};
 
-  const initialRelationObjectMetadataItem = useMemo(
-    () =>
+  const initialRelationObjectMetadataItem = useMemo(() => {
+    const availableItems = activeObjectMetadataItems.filter(
+      isObjectMetadataAvailableForRelation,
+    );
+    const initialObjectCandidate =
       relationObjectMetadataItemFromFieldMetadata ??
       objectMetadataItem ??
-      objectMetadataItems.filter(isObjectMetadataAvailableForRelation)[0],
-    [
-      objectMetadataItem,
-      objectMetadataItems,
-      relationObjectMetadataItemFromFieldMetadata,
-    ],
-  );
+      availableItems[0] ??
+      activeObjectMetadataItems[0];
+    if (!initialObjectCandidate) {
+      throw new Error(
+        'Relation Form initialization error: no valid objects to relate to. Please create a custom object in your workspace.',
```

Comment:
`Please create a custom object in your workspace.` Would replace by something like this should never occur if we're well sure this is an invariant